### ISSUE TITLE
Move dependencies into maven central

### ISSUE THREAD

**Issue by jschneidereit** [Kotlin/kotlinx-kover#34]
Since kover has something that relies on https://packages.jetbrains.team/maven/p/ij/intellij-dependencies I'm unable to use it because my company requires using an internal repository which mirrors maven central. Are there plans to make this happen?

**Comment by qwwdfsad** [comment]
Thanks for the report! We are going to do so in a nearest future

**Comment by JavierSegoviaCordoba** [comment]
Same issue I guess

```
Caused by: org.gradle.internal.resolve.ModuleVersionNotFoundException: Could not find org.jetbrains.kotlin:kotlin-scripting-compiler-embeddable:1.5.31.
Searched in the following locations:
  - https://packages.jetbrains.team/maven/p/ij/intellij-dependencies/org/jetbrains/kotlin/kotlin-scripting-compiler-embeddable/1.5.31/kotlin-scripting-compiler-embeddable-1.5.31.pom
If the artifact you are trying to retrieve can be found in the repository but without metadata in 'Maven POM' format, you need to adjust the 'metadataSources { ... }' of the repository declaration.
```

**Comment by pvegh** [comment]
0.4.0 as documented right now can't be pulled from Macen Central. Has it been uploaded yet?

**Comment by shanshin** [comment]
> 0.4.0 as documented right now can't be pulled from Macen Central. Has it been uploaded yet?

Version `0.4.0` should now be available in Maven Central.

**Comment by AshStefanOltmann** [comment]
Kover 0.4.0 worked for me using maven central

**Comment by jschneidereit** [comment]
Both 0.4.0 and 0.4.1 are at least reporting this old repository

```
An exception occurred applying plugin request [id: 'org.jetbrains.kotlinx.kover', version: '0.4.1']
> Failed to apply plugin 'org.jetbrains.kotlinx.kover'.
   > Found non-artifactory repository: https://packages.jetbrains.team/maven/p/ij/intellij-dependencies
   ```

**Comment by shanshin** [comment]
Unfortunately, the IntelliJ agent artifacts are still located only in the `packages.jetbrains.team` repository.
We plan to move them to Maven Central soon.

**Comment by JavierSegoviaCordoba** [comment]
Is there any workaround or just wait?

**Comment by JavierSegoviaCordoba** [comment]
Using this in the root `build.gradle.kts` works.

```kotlin
allprojects {
    repositories {
        mavenCentral()
    }
}
```

But using the new way to define repositories in `settings.gradle.kts` doesn't work:

```kotlin
pluginManagement {
    repositories {
        gradlePluginPortal()
        mavenCentral()
    }
}

dependencyResolutionManagement {
    repositories {
        mavenCentral()
    }
}
```

**Comment by Vampire** [comment]
@JavierSegoviaCordoba that is because the kover Gradle plugin is misbehaving in https://github.com/Kotlin/kotlinx-kover/blob/main/src/main/kotlin/kotlinx/kover/KoverPlugin.kt#L31-L33.
It adds unasked and unconfigurable a repository to the project.
This is very bad practice and for example also works against company guidelines to only use internal repositories.
If you would have set `repositoriesMode.set(RepositoriesMode.FAIL_ON_PROJECT_REPOS)` your build would even have failed due to this behavior.
You could set `repositoriesMode.set(RepositoriesMode.PREFER_SETTINGS)` and add that repo to your settings script to work-around the issue and prevent the need for `allprojects`.
And kover should be changed to **at least** provide an off-switch for that behavior.

**Comment by shanshin** [comment]
@Vampire
>that is because the kover Gradle plugin is misbehaving in https://github.com/Kotlin/kotlinx-kover/blob/main/src/main/kotlin/kotlinx/kover/KoverPlugin.kt#L31-L33.

This line will be deleted immediately when the IntelliJ coverage engine artifacts are published in the Maven Central.

**Comment by Vampire** [comment]
That's great @shanshin but still it currently would prevent one to use `repositoriesMode.set(RepositoriesMode.FAIL_ON_PROJECT_REPOS)` e. g. to enforce the exclusive usage of a corporate repository / mirror, where the artifact could already be present.
That's why I said, such code should always **at least** have an off-switch, if not even require an explicit on-switch being flipped or just require downstream projects to declare the needed repositories and just mention it in documentation. :-)

**Comment by JavierSegoviaCordoba** [comment]
Are those artifacts still missing in Maven Central?

**Comment by qwwdfsad** [comment]
The `intellij-coverage` is successfully published to Maven Central. We'll update the dependency and release new version that depends only on MC during next week

**Comment by shanshin** [comment]
Fixed in `0.4.3`
